### PR TITLE
add UPP

### PR DESCRIPTION
**Pull request by lrauschning**
Adds a module for running MSA using UPP to nf-core.
As there are some issues with the conda package for SEPP/UPP, this is drafted for now.

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by SPPearce** [comment]
Within the maintainers team we've had a discussion recently, and we want to:
-  keep the names of the meta maps as meta, meta2, meta3 etc across all modules for consistency.
- leave editing the meta maps to the workflow logic, not within modules.

File: tests/modules/nf-core/upp/align/main.nf
Comment (luisas) [inline tests/modules/nf-core/upp/align/main.nf:1]: Maybe we can add the nf-test already within this PR?

Also, i would test for the mode when a tree is used as input too :)

**Review by luisas — CHANGES_REQUESTED** [review]
Amazing @lrauschning :) 

Really good stuff!

I just added a comment about the tests, would be cool to have the nf-tests before merging.

**Comment by SPPearce** [comment]
@luisas, @lrauschning , I swapped to nf-test.
The pytest was apparently using a bam file, which didn't seem right.
Is there an appropriate test that could be make to work here, and get this module finished off?

**Comment by lrauschning** [comment]
Yeah, UPP was causing quite a few issues and I was never able to reliably get it packaged for conda (its packaged with SEPP and TIPP, and requires some file wrangling), so it got sidelined in favour of MAGUS.
I don't have the time to work on that right now, so will close the PR – if anyone else wants to work on it, they can pull from the [branch](https://github.com/lrauschning/modules/tree/UPP) in my fork of nf-core/modules and open a new PR.

**Pull request closed without merging**
